sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/DecimalArithmeticOverrides.scala

```diff
@@ -99,21 +99,36 @@ object DecimalArithmeticOverrides {
             }
         }),
 
-      /**
-       * Because of https://github.com/NVIDIA/spark-rapids/issues/7595 we are not supporting
-       * Decimals for spark 3.4 and db 11.3. Once we do we should revert the changes made to the
-       * following tests test_mod, test_mod_mixed and test_mod_pmod_by_zero_not_ansi or we should
-       * just revert this commit
-       */
       expr[Remainder](
         "Remainder or modulo",
         ExprChecks.binaryProject(
           TypeSig.gpuNumeric, TypeSig.cpuNumeric,
-          ("lhs", TypeSig.integral + TypeSig.fp, TypeSig.cpuNumeric),
-          ("rhs", TypeSig.integral + TypeSig.fp, TypeSig.cpuNumeric)),
+          ("lhs", TypeSig.gpuNumeric, TypeSig.cpuNumeric),
+          ("rhs", TypeSig.gpuNumeric, TypeSig.cpuNumeric)),
         (a, conf, p, r) => new BinaryExprMeta[Remainder](a, conf, p, r) {
+          // See https://github.com/NVIDIA/spark-rapids/issues/8330
+          // Basically if we overflow on Decimal128 values when up-casting the operands, we need
+          // to fall back to CPU since we don't currently have enough precision to support that 
+          // on the GPU. 
+          override def tagExprForGpu(): Unit = {
+            if (a.left.dataType.isInstanceOf[DecimalType] &&
+                a.right.dataType.isInstanceOf[DecimalType]) {
+              val lhsType = a.left.dataType.asInstanceOf[DecimalType]
+              val rhsType = a.right.dataType.asInstanceOf[DecimalType]
+              val needed = DecimalRemainderChecks.neededPrecision(lhsType, rhsType)
+              if (needed > DType.DECIMAL128_MAX_PRECISION) {
+                willNotWorkOnGpu(s"needed intermediate precision ($needed) will overflow " +
+                  s"outside of the maximum available decimal128 precision")
+              }
+            }
```

Comment:
```suggestion
            (a.left.dataType, a.right.dataType) match {
              
```suggestion
            if (a.left.dataType.isInstanceOf[DecimalType] &&
                a.right.dataType.isInstanceOf[DecimalType]) {
              val lhsType = a.left.dataType.asInstanceOf[DecimalType]
              val rhsType = a.right.dataType.asInstanceOf[DecimalType]
              val needed = DecimalRemainderChecks.neededPrecision(lhsType, rhsType)
              if (needed > DType.DECIMAL128_MAX_PRECISION) {
                willNotWorkOnGpu(s"needed intermediate precision ($needed) will overflow " +
                  s"outside of the maximum available decimal128 precision")
              }
            }
```case (lhsType: DecimalType, rhsType: DecimalType) =>
                val needed = DecimalRemainderChecks.neededPrecision(lhsType, rhsType)
                if (needed > DType.DECIMAL128_MAX_PRECISION) {
                  willNotWorkOnGpu(s"needed intermediate precision ($needed) will overflow " +
                    s"outside of the maximum available decimal128 precision")
                }
              case _=> ()
            }
```